packages/crayons-core/src/components/datepicker/datepicker.tsx

```diff
@@ -999,9 +999,41 @@ export class Datepicker {
 
   processValueChange(val, emitChange = false) {
     // show error if not ISO format and not display format
-    const parsedDate = parse(val, this.displayFormat, new Date(), {
-      locale: this.langModule,
-    });
+    let parsedDate;
+    if (this.langModule?.code === 'is' && this.dateFormat === 'dd MMM yyyy') {
```

Comment:
@harshith-venkatesh-freshworks Can you move this to a common method and the constants separately as we are duplicating the code here?